### ISSUE TITLE
Have utils.timeit return two elements not n elements

### ISSUE THREAD

**Issue by scottsievert** [nextml/NEXT#92]
In utils.py, utils.timeit can return a tuple with n elements instead of a tuple with two elements. We would like utils.timeit to return a tuple with two elements where the first element may be another tuple and the second element is the timing information.

This requires changing lines like `a1, a2, a3, a4, dt = utils.timeit(..)(...)` to `a, dt = utils.timeit`.

More comments can be found in this function, [utils.py:timeit][1].

[1]:https://github.com/nextml/NEXT/blob/master/next/utils/utils.py#L322

**Comment by kgjamieson** [comment]
Why is this preferred? This would always require an additional line to
parse the outputs

**Comment by scottsievert** [comment]
For the rewrite we’re using

``` python
alg_response, dt = utils.timeit(…)(…)
self.myApp.f(*alg_response)
```

which we can’t do if dt is included in the algorithm response (we don’t want to enforce that app developers include `dt` as their last argument.

On February 3, 2016 at 7:15:36 PM, Kevin Jamieson (notifications@github.com) wrote:

Why is this preferred? This would always require an additional line to
parse the outputs

**Comment by kgjamieson** [comment]
Makes sense. Thanks

**Comment by scottsievert** [comment]
Closed by b0d0b1694fa90eaf662c14b2693f1e2046e66525
